File: dsa_lab_1/1.c
#include <stdio.h>

int sum(int arr[], int n) {
    if (n == 0) {        
        return 0;
    }
    return arr[n - 1] + sum(arr, n - 1);
}

int main() {
    int n;
    int arr[10];

    printf("Enter number of elements: ");
    scanf("%d", &n);

    printf("Enter elements: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    printf("Sum = %d", sum(arr, n));
    return 0;
}

File: python_lab_12/1.py
class Student:
    def __init__(self, name, sap_id, phy, chem, maths):
        self.name = name
        self.sap_id = sap_id
        self.marks = {"Physics": phy, "Chemistry": chem, "Maths": maths}

    def display(self):
        print(f"\nName: {self.name}")
        print(f"SAP ID: {self.sap_id}")
        print(f"Marks: {self.marks}")

    def marks_percentage(self):
        total = sum(self.marks.values())
        percentage = total / 3
        return percentage

    def result(self):
        for subject, mark in self.marks.items():
            if mark <= 40:
                return "Fail"
        return "Pass"


students = []
n = int(input("Enter number of students: "))

for i in range(n):
    print(f"\nEnter details of Student {i+1}")
    name = input("Enter name: ")
    sap_id = input("Enter SAP ID: ")
    phy = float(input("Enter Physics marks: "))
    chem = float(input("Enter Chemistry marks: "))
    maths = float(input("Enter Maths marks: "))

    s = Student(name, sap_id, phy, chem, maths)
    students.append(s)

print("\n--- Student Details ---")
for s in students:
    s.display()
    print(f"Percentage: {s.marks_percentage():.2f}%")
    print(f"Result: {s.result()}")


def class_average(students):
    total = 0
    for s in students:
        total += s.marks_percentage()
    avg = total / len(students)
    return avg


print(f"\nClass Average Percentage: {class_average(students):.2f}%")